File: Interview_Prep_App/native_java/javatemplate/src/main/java/com/tc2r/javatemplate/repositories/Tc2rGithubRepository.java
package com.tc2r.javatemplate.repositories;

import com.tc2r.javatemplate.network.RetrofitBuilders;
import com.tc2r.javatemplate.repositories.models.Answer;
import com.tc2r.javatemplate.repositories.models.Question;

import java.util.ArrayList;
import java.util.Arrays;

public class Tc2rGithubRepository {
    private static final RetrofitBuilders retrofitBuilder = new RetrofitBuilders();

    //private final Tc2rGithubService tc2rGithubService = retrofitBuilder.tc2rGithubService;

    // FIXME use this function to make calls to api and convert the response
    // FIXME It is best to do this off of the main thread.
    public ArrayList<Answer> getAnswersList() {

        Answer tempAnswer = new Answer("It will perform the inter connection between activities and the data " + "passing mechanism.",
                                       "An Intent is connected to either the external world of application or " + "internal world of application ,Such as, opening a pdf is an intent and connect to the web browser" + ".etc.");

        return new ArrayList<>(Arrays.asList(tempAnswer, tempAnswer, tempAnswer, tempAnswer, tempAnswer, tempAnswer));
    }


    // FIXME use this function to make calls to api and convert the response
    // FIXME It is best to do this off of the main thread.
    public ArrayList<Question> getQuestions() {
        // val response = tc2rGithubService.getQuestionsJson()
        // val array = response.body()

        Question question = new Question.Builder(1, "multi", "What is a cat?", "felines are cats born from cats which eat cats but aren't quite cats.")
                .shortAns("it is a four legged animal that eats mice")
                .build();

        return new ArrayList<>(Arrays.asList(question, question, question, question, question, question)); // return list of questions.
    }
}

File: Interview_Prep_App/native_kotlin/template/src/main/java/com/interviewprep/template/repository/tc2rgithubrepository/Tc2rGithubRepository.kt
package com.interviewprep.template.repository.tc2rgithubrepository

import com.interviewprep.template.network.RetrofitBuilders
import com.interviewprep.template.repository.models.Answer
import com.interviewprep.template.repository.models.Question

class Tc2rGithubRepository {
    private val TAG = "ACTEST"

    companion object {
        private val retrofitBuilder: RetrofitBuilders = RetrofitBuilders()
    }

    // Creates API object using Retrofit to implement API Service
    private val tc2rGithubService = retrofitBuilder.tc2rGithubService

    // FIXME use this function to make calls to api and convert the response
    // FIXME It is best to do this off of the main thread.
    fun getTheAnswers(): ArrayList<Answer> {
        // val response = tc2rGithubService.requestAnswers()
        // val array = response.body()

        return arrayListOf(Answer(), Answer(), Answer(), Answer(), Answer()) // return list of Answers
    }

    // FIXME use this function to make calls to api and convert the response
    // FIXME It is best to do this off of the main thread.
    fun getQuestions(): ArrayList<Question> {
        // val response = tc2rGithubService.getQuestionsJson()
        // val array = response.body()

        return arrayListOf(Question(), Question(), Question(), Question(), Question()) // return list of questions.
    }
}
